src/indicator/algorithm/calculator/envelope.js

```diff
@@ -0,0 +1,40 @@
+import d3 from 'd3';
+import undefinedInputAdapter from './undefinedInputAdapter';
+
+export default function() {
+
+    var factor = 0.1;
+    var value = function(d) {
+        return d;
+    };
+    var envelopeFactor = function(valueData) {
+        return valueData.map(function(s) {
+            return {
+                lowerEnvelope: s * (1.0 - factor),
+                upperEnvelope: s * (1.0 + factor)
+            };
+        });
+    };
+    var envelope = function(data) {
+        return undefinedInputAdapter().undefinedValue({}).algorithm(envelopeFactor)(
```

Comment:
@ColinEberhardt I am not sure if this is very clean, as the algorithm will have to use value function from the calculator to decide if something is undefined or not. Is there a way around it, or is it ok?